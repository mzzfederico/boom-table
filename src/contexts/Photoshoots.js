import {createContext, useContext} from "react";

/* Our shared space for the data from the APIs for the shoots */
const PhotoshootsCtx = createContext({ data: [] });
/* A shortcut to the component providing the data down the tree */
const PhotoshootsProvider = PhotoshootsCtx.Provider;
/* A shortcut to using this data for our purposes inside the pertinent components */
const usePhotoshoots = () => useContext(PhotoshootsCtx);

export { PhotoshootsCtx, PhotoshootsProvider, usePhotoshoots };