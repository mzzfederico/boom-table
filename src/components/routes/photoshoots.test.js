/* eslint-disable no-undef */
import React from "react";
import Photoshoots from "./photoshoots";
import renderer from "react-test-renderer";

it("renders without crashing", () => {
    const tree = renderer.create(
        <Photoshoots />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});