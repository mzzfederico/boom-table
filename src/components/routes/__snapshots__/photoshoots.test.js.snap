// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="WeekTable"
>
  <div
    className="column column-monday"
  >
    monday
    <style
      jsx={true}
    >
      
                .column {
                    text-transform: lowercase;
                    font-variant: small-caps;
                    font-weight: 500;
                    font-size: 0.85rem;
                    border-bottom: 1px solid #f7f7f7;
                    margin-bottom: 1rem;
                }

                .column-monday {
                    grid-column: 1;
                }
            
    </style>
  </div>
  <div
    className="client client-monday total-top"
  >
    <div
      className="number_of_photos"
    >
      -
    </div>
    <style
      jsx={true}
    >
      
            .client-monday {
                grid-column: 1
            }

            .client p {
                font-size: 0.8rem;
                margin: 0;
            }

            .client p span {
                font-size: 1rem;
                font-weight: 700;
            }

            .number_of_photos span:nth-child(2) {
                font-size: 0.7rem;
            }
        
    </style>
  </div>
  <div
    className="column column-tuesday"
  >
    tuesday
    <style
      jsx={true}
    >
      
                .column {
                    text-transform: lowercase;
                    font-variant: small-caps;
                    font-weight: 500;
                    font-size: 0.85rem;
                    border-bottom: 1px solid #f7f7f7;
                    margin-bottom: 1rem;
                }

                .column-tuesday {
                    grid-column: 2;
                }
            
    </style>
  </div>
  <div
    className="client client-tuesday total-top"
  >
    <div
      className="number_of_photos"
    >
      -
    </div>
    <style
      jsx={true}
    >
      
            .client-tuesday {
                grid-column: 2
            }

            .client p {
                font-size: 0.8rem;
                margin: 0;
            }

            .client p span {
                font-size: 1rem;
                font-weight: 700;
            }

            .number_of_photos span:nth-child(2) {
                font-size: 0.7rem;
            }
        
    </style>
  </div>
  <div
    className="column column-wednesday"
  >
    wednesday
    <style
      jsx={true}
    >
      
                .column {
                    text-transform: lowercase;
                    font-variant: small-caps;
                    font-weight: 500;
                    font-size: 0.85rem;
                    border-bottom: 1px solid #f7f7f7;
                    margin-bottom: 1rem;
                }

                .column-wednesday {
                    grid-column: 3;
                }
            
    </style>
  </div>
  <div
    className="client client-wednesday total-top"
  >
    <div
      className="number_of_photos"
    >
      -
    </div>
    <style
      jsx={true}
    >
      
            .client-wednesday {
                grid-column: 3
            }

            .client p {
                font-size: 0.8rem;
                margin: 0;
            }

            .client p span {
                font-size: 1rem;
                font-weight: 700;
            }

            .number_of_photos span:nth-child(2) {
                font-size: 0.7rem;
            }
        
    </style>
  </div>
  <div
    className="column column-thursday"
  >
    thursday
    <style
      jsx={true}
    >
      
                .column {
                    text-transform: lowercase;
                    font-variant: small-caps;
                    font-weight: 500;
                    font-size: 0.85rem;
                    border-bottom: 1px solid #f7f7f7;
                    margin-bottom: 1rem;
                }

                .column-thursday {
                    grid-column: 4;
                }
            
    </style>
  </div>
  <div
    className="client client-thursday total-top"
  >
    <div
      className="number_of_photos"
    >
      -
    </div>
    <style
      jsx={true}
    >
      
            .client-thursday {
                grid-column: 4
            }

            .client p {
                font-size: 0.8rem;
                margin: 0;
            }

            .client p span {
                font-size: 1rem;
                font-weight: 700;
            }

            .number_of_photos span:nth-child(2) {
                font-size: 0.7rem;
            }
        
    </style>
  </div>
  <div
    className="column column-friday"
  >
    friday
    <style
      jsx={true}
    >
      
                .column {
                    text-transform: lowercase;
                    font-variant: small-caps;
                    font-weight: 500;
                    font-size: 0.85rem;
                    border-bottom: 1px solid #f7f7f7;
                    margin-bottom: 1rem;
                }

                .column-friday {
                    grid-column: 5;
                }
            
    </style>
  </div>
  <div
    className="client client-friday total-top"
  >
    <div
      className="number_of_photos"
    >
      -
    </div>
    <style
      jsx={true}
    >
      
            .client-friday {
                grid-column: 5
            }

            .client p {
                font-size: 0.8rem;
                margin: 0;
            }

            .client p span {
                font-size: 1rem;
                font-weight: 700;
            }

            .number_of_photos span:nth-child(2) {
                font-size: 0.7rem;
            }
        
    </style>
  </div>
  <div
    className="column column-saturday"
  >
    saturday
    <style
      jsx={true}
    >
      
                .column {
                    text-transform: lowercase;
                    font-variant: small-caps;
                    font-weight: 500;
                    font-size: 0.85rem;
                    border-bottom: 1px solid #f7f7f7;
                    margin-bottom: 1rem;
                }

                .column-saturday {
                    grid-column: 6;
                }
            
    </style>
  </div>
  <div
    className="client client-saturday total-top"
  >
    <div
      className="number_of_photos"
    >
      -
    </div>
    <style
      jsx={true}
    >
      
            .client-saturday {
                grid-column: 6
            }

            .client p {
                font-size: 0.8rem;
                margin: 0;
            }

            .client p span {
                font-size: 1rem;
                font-weight: 700;
            }

            .number_of_photos span:nth-child(2) {
                font-size: 0.7rem;
            }
        
    </style>
  </div>
  <div
    className="column column-sunday"
  >
    sunday
    <style
      jsx={true}
    >
      
                .column {
                    text-transform: lowercase;
                    font-variant: small-caps;
                    font-weight: 500;
                    font-size: 0.85rem;
                    border-bottom: 1px solid #f7f7f7;
                    margin-bottom: 1rem;
                }

                .column-sunday {
                    grid-column: 7;
                }
            
    </style>
  </div>
  <div
    className="client client-sunday total-top"
  >
    <div
      className="number_of_photos"
    >
      -
    </div>
    <style
      jsx={true}
    >
      
            .client-sunday {
                grid-column: 7
            }

            .client p {
                font-size: 0.8rem;
                margin: 0;
            }

            .client p span {
                font-size: 1rem;
                font-weight: 700;
            }

            .number_of_photos span:nth-child(2) {
                font-size: 0.7rem;
            }
        
    </style>
  </div>
  <style
    jsx={true}
  >
    
                    .WeekTable {
                        display: grid;
                        grid-auto-flow: column;
                        grid-template-columns: repeat(7, 1fr);
                    }

                    .WeekTable &gt; div {
                        padding: 0.5rem;
                    }

                    .WeekTable &gt; .odd {
                        background-color: #f7f7f7;
                    }

                    .WeekTable &gt; .total-bottom {
                        border-top: 2px solid #f7f7f7;
                    }

                    .WeekTable &gt; .total-top {
                        border-bottom: 2px solid #f7f7f7;
                    }
                
  </style>
</div>
`;
